devito/equation.py

```diff
@@ -82,6 +81,27 @@ def subdomain(self):
     def substitutions(self):
         return self._substitutions
 
+    @cached_property
+    def _symbolic_coefficients(self):
+        return bool(self._symbolic_functions)
+
+    @cached_property
+    def _symbolic_functions(self):
+        try:
```

Comment:
what test fails without the try-except ?
if it's "only" because sometimes we can have ``Indexed``, instead of ``Function``, then we can easily add ``symbolic_functions`` to our own ``Indexed`` [here](https://github.com/opesci/devito/blob/master/devito/types/basic.py#L974)

Reply:
Yeah, it's due to `Indexed`. In fact the test currently failing is due to and `Indexed` I believe. Will look into this now.

Reply:
Oh, it's also due to the fact that sometimes the lhs or rhs can be zero.